### PR TITLE
Add MLFlow converter component

### PR DESCRIPTION
**Pull request by novaturient95**
### Description
* Added mlflow converter component
* Added preview in the display name for model mgmt components
* Added support for stable diffusion model conversion to mlflow

### Test
* Executed locally and tested model created can be loaded using mlflow

### CODE REVIEW

File: common/components/register_model/spec.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ type: command
 
 is_deterministic: True
 
-display_name: Register model
+display_name: Register model (Public Preview)
```
Comment (elesel) [inline common/components/register_model/spec.yaml:8]: @ManojBableshwar, is this the approach you want? I don't see any Preview tags in this PR.

File: training/model_management/src/azureml/model/mgmt/processors/hftransformers/vision/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+--extra-index-url https://automlcesdkdataresources.blob.core.windows.net/finetuning-models/eval_metrics_whls/17_03_latest_eval_wheel/azureml_evaluate_mlflow-0.1.0.86769376-py3-none-any.whl
```
Comment (elesel) [inline training/model_management/src/azureml/model/mgmt/processors/hftransformers/vision/requirements.txt:1]: This should point to PyPI instead.

Comment (novaturient95) [reply training/model_management/src/azureml/model/mgmt/processors/hftransformers/vision/requirements.txt:1]: @ShipraJain01 @skasturi for above comment

File: common/components/register_model/spec.yaml (hunk 1 above)
Comment (novaturient95) [reply common/components/register_model/spec.yaml:8]: @ManojBableshwar can you please share what preview tags to add?
Should it be like:

`preview`: `True`

File: training/model_management/environments/model-management/context/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,9 @@
 azure-core=={{latest-pypi-version}}
 azure-ai-ml=={{latest-pypi-version}}
 azure-identity=={{latest-pypi-version}}
+protobuf==3.20.0
+sentencepiece==0.1.97
+accelerate==0.17.1
+datasets==2.3.2
+diffusers==0.14.0
+https://automlcesdkdataresources.blob.core.windows.net/finetuning-models/eval_metrics_whls/17_03_latest_eval_wheel/azureml_evaluate_mlflow-0.1.0.86769376-py3-none-any.whl
```
Comment (vizhur) [inline training/model_management/environments/model-management/context/requirements.txt:9]: I thought we install from private feeds/blobs for private preview. For public those should be on pypi

File: common/components/register_model/spec.yaml (hunk 1 above)
Comment (novaturient95) [reply common/components/register_model/spec.yaml:8]: Added tags:

`Preview: ""`

File: training/model_management/src/azureml/model/mgmt/processors/hftransformers/vision/requirements.txt (hunk 2 above)
Comment (skasturi) [reply training/model_management/src/azureml/model/mgmt/processors/hftransformers/vision/requirements.txt]: Done. Changed to use the PyPi one.

File: training/model_management/environments/model-management/context/requirements.txt [hunk 4]
```diff
@@ -1,3 +1,9 @@
 azure-core=={{latest-pypi-version}}
 azure-ai-ml=={{latest-pypi-version}}
 azure-identity=={{latest-pypi-version}}
+protobuf==3.20.0
+sentencepiece==0.1.97
+accelerate==0.17.1
+datasets==2.3.2
+diffusers==0.14.0
+azureml-evaluate-mlflow==0.0.1
```
Comment (ShipraJain01) [inline training/model_management/environments/model-management/context/requirements.txt:9]: please refer latest py-pi version 
azureml-evaluate-mlflow=={{latest-pypi-version}}

**Review by skasturi — APPROVED** [review]
:shipit:

File: training/model_management/environments/model-management/context/requirements.txt (hunk 3 above)
Comment (novaturient95) [reply training/model_management/environments/model-management/context/requirements.txt]: updated to use PyPi

File: training/model_management/environments/model-management/context/requirements.txt [hunk 5]
```diff
@@ -1,3 +1,9 @@
 azure-core=={{latest-pypi-version}}
 azure-ai-ml=={{latest-pypi-version}}
 azure-identity=={{latest-pypi-version}}
+protobuf==3.20.0
```
Comment (vizhur) [inline training/model_management/environments/model-management/context/requirements.txt:4]: can this environment be used for model-evaluate?

File: training/model_management/environments/model-management/context/requirements.txt [hunk 6]
```diff
@@ -1,7 +1,11 @@
 azure-core=={{latest-pypi-version}}
 azure-ai-ml=={{latest-pypi-version}}
 azure-identity=={{latest-pypi-version}}
+azureml-contrib-services=={{latest-pypi-version}}
```
Comment (vizhur) [inline training/model_management/environments/model-management/context/requirements.txt:4]: what's in contrib services, I thought that was deprecated long time ago

Comment (novaturient95) [reply training/model_management/environments/model-management/context/requirements.txt:4]: still being released: https://pypi.org/project/azureml-contrib-services/ here. Let me know if I need to remove this. 
Its used in the scoring script only

Comment (novaturient95) [reply training/model_management/environments/model-management/context/requirements.txt]: Removed. @vizhur please help to check

Comment (skasturi) [reply training/model_management/environments/model-management/context/requirements.txt]: Let us get first set of models to Prod and then handle the next set that require these new packages.

File: training/model_management/components/convert_model_to_mlflow/asset.yaml
Comment (skasturi) [inline training/model_management/components/convert_model_to_mlflow/asset.yaml:1]: Add new line at the end of the file

File: training/model_management/components/convert_model_to_mlflow/spec.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+command: >
+  python run_model_preprocess.py
+  $[[--model-id ${{inputs.model_id}}]]
+  $[[--task-name ${{inputs.task_name}}]]
+  $[[--model-download-metadata ${{inputs.model_download_metadata}}]]
+  $[[--license-file-path ${{inputs.license_file_path}}]]
+  --mlflow-flavor transformers
+  --model-path ${{inputs.model_path}}
+  --mlflow-model-output-dir ${{outputs.mlflow_model_folder}}
+  --model-job-path ${{outputs.model_job_path}}
```
Comment (skasturi) [inline training/model_management/components/convert_model_to_mlflow/spec.yaml:24]: rename to `model-import-job-path`?

File: training/model_management/environments/model-management/context/requirements.txt [hunk 8]
```diff
@@ -2,6 +2,9 @@ azure-core=={{latest-pypi-version}}
 azure-ai-ml=={{latest-pypi-version}}
 azure-identity=={{latest-pypi-version}}
 azureml-evaluate-mlflow=={{latest-pypi-version}}
+ffmpeg=={{latest-pypi-version}}
```
Comment (skasturi) [inline training/model_management/environments/model-management/context/requirements.txt:5]: Could we please move these changes to a separate PR?

File: training/model_management/src/azureml/model/mgmt/processors/preprocess.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+# Licensed under the MIT License.
+
+"""Preprocess model."""
+
+import azureml.model.mgmt.processors.transformers as transformers
+from azureml.model.mgmt.config import ModelFlavor
+from pathlib import Path
+
+
+def run_preprocess(model_flavor: str, model_path: Path, output_dir: Path, **preprocess_args):
```
Comment (skasturi) [inline training/model_management/src/azureml/model/mgmt/processors/preprocess.py:11]: Please add docs

File: training/model_management/environments/model-management/context/requirements.txt (hunk 8 above)
Comment (novaturient95) [reply training/model_management/environments/model-management/context/requirements.txt]: Sure. Kept langcodes as its required for download component

File: training/model_management/src/azureml/model/mgmt/processors/transformers/convert.py [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    hf_conf['force_load_config'] = False
+    return {
+        "hf_model": model,
+        "hf_conf": hf_conf,
+        "path": output_dir,
+        "code_paths": [predict]
+    }
+
+
+def _get_whisper_model_to_save(input_dir: Path, output_dir: Path, hf_conf: Dict = {}) -> Dict:
```
Comment (skasturi) [inline training/model_management/src/azureml/model/mgmt/processors/transformers/convert.py:127]: Move out to separate PR?

File: training/model_management/src/azureml/model/mgmt/processors/transformers/diffusion/predict.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+def _load_pyfunc(path):
+    """Load pyfunc flavour."""
+    return DiffusionPyFunc(path)
+
+
+class DiffusionPyFunc:
+    """StableDiffusion pipeline."""
+
+    def __init__(self, model_path):
```
Comment (skasturi) [inline training/model_management/src/azureml/model/mgmt/processors/transformers/diffusion/predict.py:21]: Please add docs

File: training/model_management/environments/model-management/context/requirements.txt [hunk 12]
```diff
@@ -2,6 +2,7 @@ azure-core=={{latest-pypi-version}}
 azure-ai-ml=={{latest-pypi-version}}
 azure-identity=={{latest-pypi-version}}
 azureml-evaluate-mlflow=={{latest-pypi-version}}
+langcodes=={{latest-pypi-version}}
```
Comment (skasturi) [inline training/model_management/environments/model-management/context/requirements.txt:5]: Is this needed for existing set of packages?

File: training/model_management/src/azureml/model/mgmt/processors/transformers/convert.py (hunk 10 above)
Comment (novaturient95) [reply training/model_management/src/azureml/model/mgmt/processors/transformers/convert.py]: Removed whisper changes

File: training/model_management/src/azureml/model/mgmt/processors/preprocess.py (hunk 9 above)
Comment (novaturient95) [reply training/model_management/src/azureml/model/mgmt/processors/preprocess.py]: Added. Please check

File: training/model_management/src/azureml/model/mgmt/processors/transformers/diffusion/predict.py (hunk 11 above)
Comment (novaturient95) [reply training/model_management/src/azureml/model/mgmt/processors/transformers/diffusion/predict.py]: Added

File: training/model_management/environments/model-management/context/requirements.txt (hunk 12 above)
Comment (novaturient95) [reply training/model_management/environments/model-management/context/requirements.txt:5]: required for download component

**Review by skasturi — APPROVED** [review]

**Comment by skasturi** [comment]
Tested successfuly on WS registry: 

image

**Review by skasturi — APPROVED** [review]

**Comment by skasturi** [comment]
"common" changes moved to a separate PR: https://github.com/Azure/azureml-assets/pull/486
